src/mlpack/core/util/facilities.hpp

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file facilities.hpp
```

Comment:
To be honest, the name `facilities.hpp` isn't very descriptive.  Do you think that we could change it to something like `size_checks.hpp` or some other name that makes it more clear what the function is for?

Reply:
I too agree with you. `size_checks.hpp` is definitely far more descriptive than the current one :)
Will update it.